packages/cli/test/mocks/client.ts

```diff
@@ -112,6 +153,41 @@ export class MockClient extends Client {
     this.cwd = originalCwd;
   }
 
+  events = {
+    keypress(
+      key:
+        | string
+        | {
+            name?: string | undefined;
+            ctrl?: boolean | undefined;
+            meta?: boolean | undefined;
+            shift?: boolean | undefined;
+          }
+    ) {
+      if (typeof key === 'string') {
+        client.stdin.emit('keypress', null, { name: key });
+      } else {
+        client.stdin.emit('keypress', null, key);
+      }
+    },
+    type(text: string) {
+      client.stdin.write(text);
+      for (const char of text) {
+        client.stdin.emit('keypress', null, { name: char });
+      }
+    },
+  };
+
+  getScreen({ raw }: { raw?: boolean } = {}): string {
+    const stderr = client.stderr as any as MockStream;
+    const lastScreen = stderr.getLastChunk({ raw });
+    return raw ? lastScreen : stripAnsi(lastScreen).trim();
+  }
+  getFullOutput(): string {
+    const stderr = client.stderr as any as MockStream;
+    return stderr.getFullOutput();
+  }
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** if we can remove type assertions, that'd be great

When I remove this locally, it seems to work fine.

```suggestion
  getScreen({ raw }: { raw?: boolean } = {}): string {
    const lastScreen = client.stderr.getLastChunk({ raw });
    return raw ? lastScreen : stripAnsi(lastScreen).trim();
  }
  getFullOutput(): string {
    return client.stderr.getFullOutput();
  }
```

Reply:
Yes makes sense. I think I was trying to do some type fixes to the MockClient (that didn't work) and this was some leftover code.

Reply:
Regarding the type errors, these are kind of everywhere and have been for a little while as far as I know. Not related to this PR, unless you're seeing something different. I don't think there's a great solution to fix this, have a [draft PR here](https://github.com/vercel/vercel/pull/11345) that I've been meaning to revisit.